### PR TITLE
[NPM-3047] Fix negative tests and test handling in flaky TestReadInitialUDPState test

### PR DESCRIPTION
**Pull request by AyyLam**
### What does this PR do?
This PR fixes the flaky behavior from https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/360112683 by using `require.EventuallyWithT` with assertions instead of `t.Errorf` and checking a hardcoded port is unused (not being listened on) before running negative tests on it. 

### Motivation

https://datadoghq.atlassian.net/browse/NPM-3047

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/port_test.go [hunk 1]
```diff
@@ -155,24 +156,25 @@ func TestReadInitialUDPState(t *testing.T) {
 		require.NoError(t, err)
 		for _, p := range ports[:2] {
 			if _, ok := initialPorts[PortMapping{testRootNs, p}]; !ok {
-				t.Errorf("PortMapping(testRootNs, p) returned false for port %d", p)
+				t.Logf("PortMapping(testRootNs, p) returned false for port %d", p)
```
Comment (hmahmood) [inline pkg/network/port_test.go:159]: Can we actually rewrite these bits using `assert.Contains`? Replacing with `Logf` will not fail the test eventually which is what we here.

File: pkg/network/port_test.go [hunk 2]
```diff
@@ -190,3 +192,11 @@ func getPort(t *testing.T, listener net.Listener) uint16 {
 func getPortUDP(_ *testing.T, udpConn *net.UDPConn) uint16 {
 	return uint16(udpConn.LocalAddr().(*net.UDPAddr).Port)
 }
+
+func isUnusedUDPPort(_ *testing.T, port int) bool {
+	_, err := net.ListenUDP("udp", &net.UDPAddr{Port: port})
+	if err != nil && strings.Contains(err.Error(), "address already in use") {
+		return false
+	}
+	return true
```
Comment (hmahmood) [inline pkg/network/port_test.go:201]: We should close the listener above in the case of success. You can do that by calling `t.Cleanup` here.

File: pkg/network/port_test.go (hunk 1 above)
Comment (AyyLam) [reply pkg/network/port_test.go]: I decided for us to use `assert.EventuallyWithT` so we can freely use assertions and not deal with the boolean return.

File: pkg/network/port_test.go (hunk 2 above)
Comment (AyyLam) [reply pkg/network/port_test.go:194]: Fixed.

File: pkg/network/port_test.go [hunk 3]
```diff
@@ -150,32 +152,22 @@ func TestReadInitialUDPState(t *testing.T) {
...
 		}
 		for _, p := range ports[2:] {
-			if _, ok := initialPorts[PortMapping{nsIno, p}]; !ok {
-				t.Errorf("PortMapping(nsIno, p) returned false for port %d", p)
-				return false
-			}
+			assert.Contains(t, initialPorts, PortMapping{nsIno, p}, fmt.Sprintf("PortMapping(nsIno, p) returned false for port %d", p))
 		}
+		if isUnusedUDPPort(t, 999) {
+			_, ok := initialPorts[PortMapping{testRootNs, 999}]
```
Comment (hmahmood) [inline pkg/network/port_test.go:165]: You can use `assert.NotContains` here (and below)

File: pkg/network/port_test.go [hunk 4]
```diff
@@ -150,32 +152,22 @@ func TestReadInitialUDPState(t *testing.T) {
 	nsIno, err := kernel.GetInoForNs(ns)
 	require.NoError(t, err)
 
-	require.Eventually(t, func() bool {
+	require.EventuallyWithT(t, func(c *assert.CollectT) {
 		initialPorts, err := ReadInitialState("/proc", UDP, true)
 		require.NoError(t, err)
```
Comment (hmahmood) [inline pkg/network/port_test.go:157]: Should use `c` here as well

File: pkg/network/port_test.go [hunk 5]
```diff
@@ -150,32 +152,19 @@ func TestReadInitialUDPState(t *testing.T) {
...
-
-		if _, ok := initialPorts[PortMapping{testRootNs, 999}]; ok {
-			t.Errorf("expected IsListening(testRootNs, 999) to return false, but returned true")
-			return false
+			assert.Contains(c, initialPorts, PortMapping{nsIno, p}, fmt.Sprintf("PortMapping(nsIno, p) returned false for port %d", p))
 		}
-		if _, ok := initialPorts[PortMapping{nsIno, 999}]; ok {
-			t.Errorf("expected IsListening(testRootNs, 999) to return false, but returned true")
-			return false
+		if isUnusedUDPPort(t, 53) {
```
Comment (hmahmood) [inline pkg/network/port_test.go:164]: Port `53` is very likely to be in use as there is a separate change in the works to have a local dns server running.

File: pkg/network/port_test.go (hunk 3 above)
Comment (AyyLam) [reply pkg/network/port_test.go]: 👍 Ah thank you, I don't know how I missed this when was trying to find the opposite of `assert.Contains`.

File: pkg/network/port_test.go (hunk 5 above)
Comment (AyyLam) [reply pkg/network/port_test.go:164]: 🤦 I was using 53 for testing. Thanks for catching. I'll switch back to 999

**Review by hmahmood — APPROVED** [review]

**Pull request merged**